infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/logreader/StreamsLogEntryReader.java

```diff
@@ -124,6 +131,46 @@ private LogReplicationEntryMsg generateMessageWithOpaqueEntryList(
         return txMessage;
     }
 
+    private void checkStreams(Set<UUID> txEntryStreamIds) {
+        // Get the set of stream ids that we encounter for the first time. i.e. Neither recorded
+        // in LogReplicationConfig nor marked as noisy.
+        Set<UUID>  checkIdSet = txEntryStreamIds.stream()
+                .filter(id -> !streamUUIDs.contains(id) && !confirmedNoisyStreams.contains(id))
+                .collect(Collectors.toSet());
+
+        if (checkIdSet.isEmpty()) {
+            return;
+        }
+
+        // TODO: why not get check name set directly?
+        CorfuTable<CorfuStoreMetadata.TableName,
+                CorfuRecord<CorfuStoreMetadata.TableDescriptors, CorfuStoreMetadata.TableMetadata>>
+                registryTable = runtime.getTableRegistry().getRegistryTable();
+
+        Set<CorfuStoreMetadata.TableName> checkNameSet = registryTable.keySet().stream()
```

Comment:
if a stream id is present in the transaction entry, it will have been opened and hence guaranteed to be in the registry table.  So this check and filter are not needed.

Reply:
Thanks for pointing out I got it. The reason we create a `checkNameSet` here is we need the table name to get the table metadata from RegistryTable and further get the `is_federated` flag. And the code above only uses UUID to track streams. But yeah I agree the code here is a little redundant and we can do some improvement here.